File: src/main/java/com/github/kayjamlang/core/expressions/VariableExpression.java
package com.github.kayjamlang.core.expressions;

import com.github.kayjamlang.core.opcodes.AccessType;

public class VariableExpression extends Expression {
    /**
     * Variable name
     */
    public final String name;

    /**
     * Set value
     */
    public final Expression expression;

    /**
     * @param name Variable name
     * @param expression Set value
     * @param accessType Type of access
     * @param line Start line
     */
    public VariableExpression(String name, Expression expression, AccessType accessType, int line) {
        super(accessType, line);
        this.name = name;
        this.expression = expression;
    }
}

File: src/main/scala/com/github/kayjamlang/core/expressions/VariableExpression.scala
package com.github.kayjamlang.core.expressions

import com.github.kayjamlang.core.Type
import com.github.kayjamlang.core.opcodes.AccessType

class VariableExpression(val name: String,
                         val expression: Expression,
                         val variableType: Type = null,
                         accessType: AccessType,
                         line: Int) extends Expression(accessType, line)